### ISSUE TITLE
can.Map.define "Use" documentation/extending define

### ISSUE THREAD

**Issue by jandjorgensen** [bitovi/canjs#1381]
Overall, the "use" for can.Map.define should probably look a little more like a guide.

The documentation should clarify define/extend behavior so this is obvious:
```
var A = can.Map.extend({
    define: { foo: { ... } } 
});

var B = A.extend({
    define: { bar: { ... } }
});

// only B's define gets used
var c = new B({});
c.attr('foo') // doesn't use A's attrDefinitions
```
Could also include instructions on using `can.extend` in `init` to extend the `define` object.

**Comment by jandjorgensen** [comment]
Fixed by #1346 but should probably be documented

**Comment by jandjorgensen** [comment]
Seems to work correctly these days--no longer sure what this issue was actually about anyway.
